src/Microsoft.SqlTools.ServiceLayer/QueryExecution/DataStorage/SaveAsMarkdownFileStreamFactory.cs

```diff
@@ -0,0 +1,68 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.SqlTools.ServiceLayer.QueryExecution.Contracts;
+using Microsoft.SqlTools.ServiceLayer.SqlContext;
+using Microsoft.SqlTools.ServiceLayer.Utility;
+
+namespace Microsoft.SqlTools.ServiceLayer.QueryExecution.DataStorage
+{
+    public class SaveAsMarkdownFileStreamFactory : IFileStreamFactory
+    {
+        private readonly SaveResultsAsMarkdownRequestParams _saveRequestParams;
+
+        /// <summary>
+        /// Constructs and initializes a new instance of <see cref="SaveAsMarkdownFileStreamFactory"/>.
+        /// </summary>
+        /// <param name="requestParams">Parameters for the save as request</param>
+        public SaveAsMarkdownFileStreamFactory(SaveResultsAsMarkdownRequestParams requestParams)
+        {
+            this._saveRequestParams = requestParams;
+        }
+
+        /// <inheritdoc />
+        public QueryExecutionSettings QueryExecutionSettings { get; set; }
+
+        /// <inheritdoc />
+        /// <exception cref="InvalidOperationException">Throw at all times.</exception>
+        [Obsolete("Not implemented for export factories.")]
```

Comment:
Let's put this in the exception message instead 

`InvalidOperationException("CreateFile not implemented for export factories");`